ray_on_golem/server/services/new_golem.py

```diff
@@ -0,0 +1,117 @@
+import json
+import logging
+from datetime import datetime, timedelta, timezone
+from decimal import Decimal
+from pathlib import Path
+from typing import Optional
+
+from golem.managers import DefaultPaymentManager
+from golem.node import GolemNode
+from golem.resources import Allocation, AllocationException, DeployArgsType, Network
+from golem.utils.logging import trace_span
+from ya_payment import models
+from yarl import URL
+
+from ray_on_golem.server.services.golem.helpers.demand_config import DemandConfigHelper
+from ray_on_golem.server.settings import YAGNA_PATH
+from ray_on_golem.utils import run_subprocess_output
+
+logger = logging.getLogger(__name__)
+
+DEFAULT_DEMAND_LIFETIME = timedelta(hours=8)
+DEFAULT_LONG_RUNNING_DEMAND_LIFETIME = timedelta(days=365)
+DEFAULT_DEBIT_NOTE_INTERVAL = timedelta(minutes=3)
+DEFAULT_DEBIT_NOTES_ACCEPT_TIMEOUT = timedelta(minutes=4)
+DEFAULT_PROPOSAL_RESPONSE_TIMEOUT = timedelta(seconds=30)
+DEFAULT_SSH_SENTRY_TIMEOUT = timedelta(minutes=2)
+DEFAULT_MAX_SENTRY_FAILS_COUNT = 3
+
+EXPIRATION_TIME_FACTOR = 0.8
+
+
+class NoMatchingPlatform(AllocationException):
+    ...
+
+
+# FIXME: Rework this into the golem-core's DefaultPaymentManager and Allocation on yagna 0.16+,
+#  as until then there is no api call available to get driver lists and golem-core is api-only
+class DeviceListAllocationPaymentManager(DefaultPaymentManager):
```

Comment:
what's "DeviceList" here? please add a docstring to this class...

Reply:
It should be `DriverList`.